RocketChat/Rocket.Chat
# feat: Use `application/octet-stream` as a fallback media type to avoid "Unknown media type" errors

## Proposed changes (including videos or screenshots)

 - Allow any file with an unknown media type (the ones not defined in IANA, Apache or nginx MIME types will be classified as `application/octet-stream` as default if the new setting is disabled) to be uploaded by using a `application/octet-stream` as the default/fallback MIME type;
 - Normalize how MIME types are identified throughout the application (by using the `mime` package in the back-end too)
 - Add alert to "Blocked Media Types" setting
![Captura de tela de 2024-06-17 13-15-45](https://github.com/RocketChat/Rocket.Chat/assets/36537004/61ff0516-60cd-4148-a2f1-b076dcb96e4f)

## Issue(s)


## Steps to test or reproduce

Access a room and try uploading a `drawio` file in it (such as the one available in the end-to-end tests added by this PR); **_Expected behavior:_** The file should be accepted and uploaded succesfully. Check the `rocketchat_uploads` DB collection to confirm the file has been uploaded with the `application/octet-stream` media type (default one for unknown media extensions);
**_Current behavior:_** the file isn't accepted and an "Unknown media type not accepted" error is thrown.

## Further comments

[SUP-531](https://rocketchat.atlassian.net/browse/SUP-531)


[SUP-531]: https://rocketchat.atlassian.net/browse/SUP-531?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
